src/ray/core_worker/lib/java/jni_helper.h

```diff
@@ -0,0 +1,174 @@
+#ifndef RAY_COMMON_JAVA_JNI_HELPER_H
+#define RAY_COMMON_JAVA_JNI_HELPER_H
+
+#include <jni.h>
+#include "ray/common/buffer.h"
+#include "ray/common/id.h"
+#include "ray/common/status.h"
+#include "ray/core_worker/store_provider/store_provider.h"
+
+/// Boolean class
+extern jclass java_boolean_class;
+/// Constructor of Boolean class
+extern jmethodID java_boolean_init;
+
+/// List class
+extern jclass java_list_class;
+/// size method of List class
+extern jmethodID java_list_size;
+/// get method of List class
+extern jmethodID java_list_get;
+/// add method of List class
+extern jmethodID java_list_add;
+
+/// ArrayList class
+extern jclass java_array_list_class;
+/// Constructor of ArrayList class
+extern jmethodID java_array_list_init;
+/// Constructor of ArrayList class with single parameter capacity
+extern jmethodID java_array_list_init_with_capacity;
+
+/// RayException class
+extern jclass java_ray_exception_class;
+
+/// NativeRayObject class
+extern jclass java_native_ray_object_class;
+/// Constructor of NativeRayObject class
+extern jmethodID java_native_ray_object_init;
+/// data field of NativeRayObject class
+extern jfieldID java_native_ray_object_data;
+/// metadata field of NativeRayObject class
+extern jfieldID java_native_ray_object_metadata;
+
+/// Throws a Java RayException if the status is not OK.
+inline bool ThrowRayExceptionIfNotOK(JNIEnv *env, const ray::Status &status) {
```

Comment:
We can change `ThrowRayExceptionIfNotOK` to a marco to make sure the caller function always return when not ok.

Reply:
But to return what since we don't know the return type of caller function.

Reply:
the return value can be a parameter of the macro

Reply:
Done.